include/deal.II/integrators/advection.h

```diff
@@ -196,6 +202,84 @@ namespace LocalIntegrators
     }
 
     /**
+     * Advection residual operator in weak form
+     *
+     * \f[ r_i = \int_Z  (\mathbf w \cdot \nabla)v\, u_i \, dx. \f]
+     */
+    template <int dim>
+    inline void
+    cell_residual  (
+      Vector<double> &result,
+      const FEValuesBase<dim> &fe,
+      const std::vector<double> &input,
+      const VectorSlice<const std::vector<std::vector<double> > > &velocity,
+      double factor = 1.)
+    {
+      const unsigned int nq = fe.n_quadrature_points;
+      const unsigned int n_dofs = fe.dofs_per_cell;
+      Assert(input.size() == nq, ExcDimensionMismatch(input.size(), nq));
+      Assert(result.size() == n_dofs, ExcDimensionMismatch(result.size(), n_dofs));
+
+      AssertDimension(velocity.size(), dim);
+      const unsigned int v_increment = (velocity[0].size() == 1) ? 0 : 1;
```

Comment:
If I read this line correctly it means that velocity can either be constant on the cell (velocity[0].size()==1) or be defined by the user in each quadrature point. If it is not documented clearly in the header of this file, please do so here.